### DISCUSSION TITLE
What are the options for authentication?

### DISCUSSION THREAD

**Discussion by jean** [open-telemetry/opentelemetry-collector#2429] [Q&A]
Hi, how would auth work? I saw this [authenticationprocessor](https://github.com/observatorium/observatorium-otelcol/tree/ab7ac790b11e0e21bd36f6afd69ac781366a794c/processors/authenticationprocessor) being referenced, but it isn't in master, and it looks like they're using [oidc](https://github.com/observatorium/observatorium-otelcol/blob/master/deploy/observatorium-otelcol.yaml#L23) instead. 

I found [this blog post](https://medium.com/opentelemetry/securing-your-opentelemetry-collector-1a4f9fa5bd6f) on OIDC auth. Are there more docs on the topic?
Are there other recommended options?

[the remark this answers is not in the document]
**Reply by jpkrohling** [reply]
Thanks for tagging me!

**Comment by jpkrohling** [comment]
That blog post is the most comprehensive doc at the moment. The authentication processor was an earlier iteration of the work in this area (#1728), and the final solution was a set of helpers that should work with any receivers (#1810). From what I remember, we have auth implemented only for gRPC receivers, and support for HTTP can be added if there's demand. OIDC is the only supported authenticator at the moment, but support for other ones is being discussed (#2471, #2229).

Do you have a specific question about it? What are you trying to accomplish?

**Reply by Depechie** [reply]
@jpkrohling sorry to hijack this thread... but I was looking for some guidance on how to have the OIDC auth working on the receiver side together with Azure AD. ( https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc )
Is there some sample for that somewhere?

**Reply by jpkrohling** [reply]
We have an OIDC Auth Extension: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/oidcauthextension

I wrote a blog post some years ago on how to use it with Keycloak, I remember someone did it on Azure as well based on that. Try searching for issues on this repo on that, I remember there was as small change necessary to get it working.

https://medium.com/opentelemetry/securing-your-opentelemetry-collector-1a4f9fa5bd6f
